hscontrol/policy/v2/types.go

```diff
@@ -409,13 +414,44 @@ func (ag *AutoGroup) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
-func (ag AutoGroup) Resolve(_ *Policy, _ types.Users, _ types.Nodes) (*netipx.IPSet, error) {
+func (ag AutoGroup) Resolve(p *Policy, users types.Users, nodes types.Nodes) (*netipx.IPSet, error) {
+	var build netipx.IPSetBuilder
+
 	switch ag {
 	case AutoGroupInternet:
 		return util.TheInternet(), nil
-	}
 
-	return nil, nil
+	case AutoGroupMember:
+		// autogroup:member represents all untagged devices in the tailnet.
+		for _, node := range nodes {
+			if len(node.ForcedTags) != 0 {
+				continue
+			}
+			if node.Hostinfo != nil && len(node.Hostinfo.RequestTags) != 0 {
+				continue
+			}
+			node.AppendToIPSet(&build)
+		}
+
+		return build.IPSet()
+
+	case AutoGroupTagged:
+		// autogroup:tagged represents all devices with a tag in the tailnet.
+		for _, node := range nodes {
+			if len(node.ForcedTags) != 0 {
+				node.AppendToIPSet(&build)
+				continue
+			}
+			if node.Hostinfo != nil && len(node.Hostinfo.RequestTags) != 0 {
+				node.AppendToIPSet(&build)
+			}
```

Comment:
So this one is a bit more complicated, these are the tags that the node requests, but that doesnt mean its allowed to have them. These has to be checked against the policy to see of the node is allowed to selfassign the tag, this is done in here:

https://github.com/juanfont/headscale/blob/main/hscontrol/policy/v2/types.go#L213

And as you can see by the lengthy comment, the plan is to rewrite that a bit (probably in next release), so you could first just copy that and then I'll fix it up at the same time.